app/src/main/java/com/example/internetshop/presentation/adapters/ProductsCategoryAdapter.kt

```diff
@@ -0,0 +1,39 @@
+package com.example.internetshop.presentation.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.example.internetshop.databinding.ItemProductCategoryBinding
+import com.example.internetshop.domain.data.model.product.SimpleProduct
+import com.example.internetshop.presentation.adapters.holder.ProductCategoryViewHolder
+
+class ProductsCategoryAdapter(val clickListener: (SimpleProduct) -> Unit) :
+    RecyclerView.Adapter<ProductCategoryViewHolder>() {
+    var productList: MutableList<SimpleProduct> = mutableListOf()
```

Comment:
To follow incapsulation principle this list should be private then no one can modify it

Reply:
fixed